From 4ff69118c915b0c8349ea63df0013c79c6db8969 Mon Sep 17 00:00:00 2001
From: Bart Ribbers <bribbers@disroot.org>
Date: Sat, 27 Feb 2021 21:36:53 +0100
Subject: [PATCH] CMake: add version info to library

This was done with autotools (although wrongly, version was set to
0.0.0) but didn't happen with CMake yet.
This also helps distribution packaging when they split out development
files from the main package
---
 CMakeLists.txt | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8c492fc..e74b077 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,5 +1,5 @@
 cmake_minimum_required(VERSION 3.5 FATAL_ERROR)
-project(prime_server LANGUAGES CXX C)
+project(prime_server LANGUAGES CXX C VERSION 0.6.7)
 INCLUDE(FindPkgConfig)
 
 ## Get version
@@ -89,7 +89,10 @@ target_link_libraries(prime_proxyd prime_server ${CMAKE_THREAD_LIBS_INIT})
 add_executable(prime_workerd ${CMAKE_SOURCE_DIR}/src/prime_workerd.cpp)
 target_link_libraries(prime_workerd prime_server ${CMAKE_THREAD_LIBS_INIT})
 
-set_target_properties(prime_server PROPERTIES PUBLIC_HEADER "${PRIME_LIBRARY_HEADERS}")
+set_target_properties(prime_server PROPERTIES
+	PUBLIC_HEADER "${PRIME_LIBRARY_HEADERS}"
+	SOVERSION ${PROJECT_VERSION_MAJOR}
+	VERSION ${PROJECT_VERSION})
 
 # Install paths
 set(prefix "${CMAKE_INSTALL_PREFIX}")
